demos/ios/MASVS-AUTH/MASTG-DEMO-0044/script.js

```diff
@@ -0,0 +1,66 @@
+
+const AccessControlFlags = {
+    kSecAccessControlUserPresence: 1 << 0,
+    kSecAccessControlBiometryAny: 1 << 1,
+    kSecAccessControlBiometryCurrentSet: 1 << 3,
+    kSecAccessControlDevicePasscode: 1 << 4,
+    kSecAccessControlWatch: 1 << 5,
+    kSecAccessControlOr: 1 << 14,
+    kSecAccessControlAnd: 1 << 15,
+    kSecAccessControlPrivateKeyUsage: 1 << 30,
+    kSecAccessControlApplicationPassword: 1 << 31,
+  };
+
+
+Interceptor.attach(Module.getExportByName(null, 'SecAccessControlCreateWithFlags'), {
+    /* 
+        func SecAccessControlCreateWithFlags(
+        _ allocator: CFAllocator?,
+        _ protection: CFTypeRef,
+        _ flags: SecAccessControlCreateFlags,
+        _ error: UnsafeMutablePointer<Unmanaged<CFError>?>?
+        )  -> SecAccessControl?
+    */
+  onEnter(args) {
+    const flags = args[2]
+    const flags_description = parseAccessControlFlags(flags)
+    console.log(`\SecAccessControlCreateWithFlags(..., 0x${flags.toString(16)}) called with ${flags_description}\n`)
+    
+
+    if(flags == AccessControlFlags.kSecAccessControlUserPresence || 
+        // e.g. SecAccessControlCreateFlags = [.userPresence]
+        ((flags & AccessControlFlags.kSecAccessControlDevicePasscode) && (flags & AccessControlFlags.kSecAccessControlOr)) ||
+        // e.g. SecAccessControlCreateFlags = [.biometryAny, .or, .devicePasscode]
+        ((flags & AccessControlFlags.kSecAccessControlUserPresence) && (flags & AccessControlFlags.kSecAccessControlOr))
+        // e.g. SecAccessControlCreateFlags = [.biometryAny, .or, .userPresence]
+    ){
+        console.log("[WARNING] The authentication may fallback to device's passcode!")
+    }
+    else{
+        console.log("There are no flags that fall back to device's passcode")
+    }
+
```

Comment:
We can remove this so that the script can be reused for other purposes. In this case this script and the one in demo 0046 become the same script, right? That'd be great.

```suggestion
```